File: 要求二.py
#Reference:https://sites.google.com/site/ezpythoncolorcourse/dict、 https://stackoom.com/question/3yOX7/%E5%A6%82%E4%BD%95%E5%AF%B9%E5%AD%97%E5%85%B8%E4%B8%AD%E7%9A%84%E5%80%BC%E6%B1%82%E5%B9%B3%E5%9D%87%E5%80%BC
#python的list，符號為[]
#python的dic，符號為{}
#employees是list

def avg(data):
# 請用你的程式補完這個函式的區塊
    salarySum = 0
    salaryAvg = 0
    headCount = 0
    employeesList = data['employees']
    for i in range(0, int(data['count'])):
        headCount += 1              #算出count是多少，也就是人數多少
 
    for j in range(0,headCount):    #<class 'list'>  
        each = (employeesList[j])
        eachSalary = each['salary'] #每個name的salary
        salarySum += eachSalary
    j+=1                        #salarySum是所有人salary的加總

    salaryAvg = salarySum / headCount
    print('Average Salary: '+str(salaryAvg)) 

avg({
"count":3,
"employees":[
{
"name":"John",
"salary":30000
},
{
"name":"Bob",
"salary":60000
},
{
"name":"Jenny",
"salary":50000
}
]
}) # 呼叫 avg 函式



# 呼叫 avg 函式

#字典
                                #key            #key 
#data={"count":3, "employees":[{"name":"John", "salary":30000}, {"name":"Bob", "salary":60000}, {"name":"Jenny", "salary":50000}]}
#data['employees']是一個list名稱
#data['employees'][0] 抓第一組
#data['employees'][1] 抓第二組
#data['employees'][2] 抓第三組
#print(data['count'])     #取出3
#employeesList = data['employees']
#print(data['employees']) #取字典內容要用[]
#print(type(employeesList))    #<class 'list'>
#print(employeesList[0])       #{'name': 'John', 'salary': 30000}
#print(type(employeesList[0])) #<class 'dict'>

File: 要求二.js
//Reference: https://developer.mozilla.org/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Array
//Reference: https://ccckmit.gitbooks.io/javascript/content/ch5/chapter5.html
//Reference: https://ccckmit.gitbooks.io/javascript/content/ch3/chapter3.html
//let 與 var不同之處：https://realdennis.medium.com/%E9%96%92%E8%81%8A-var%E8%88%87let%E7%9A%84%E6%87%B6%E4%BA%BA%E5%8C%85-javascript-b5a3f40ee28d
function avg(data){
    // 請用你的程式補完這個函式的區塊
    var salarySum = 0;
    var salaryAvg = 0;
    var headCount = 0;
    var employeesList = data.employees;    //data是字典，要取出要使用data.employees
    for(i=0; i<data.count; i++){           //data是字典，要取出3要使用data.count
        headCount = headCount + 1
    };
    for(j=0; j<headCount; j++){
        eachSalary = employeesList[j].salary;
        salarySum += eachSalary;
    }
    j+=1;
    salaryAvg = salarySum / headCount;
    document.write(salaryAvg)
}

avg({
"count":3,
"employees":[
{
"name":"John",
"salary":30000
},
{
"name":"Bob",
"salary":60000
},
{
"name":"Jenny",
"salary":50000
}
]
}); // 呼叫 avg 函式



//employees是個array
//var employees = [{"name":"John","salary":30000}, {"name":"Bob", "salary":60000}, {"name":"Jenny","salary":50000}]
//跑出打印結果
//console.log(employees)
/*[
    { name: 'John', salary: 30000 },
    { name: 'Bob', salary: 60000 },
    { name: 'Jenny', salary: 50000 }
  ]
*/
//console.log(employees[0])
//{ name: 'John', salary: 30000 }

//console.log(employees[0].salary)
//30000